File: carddemo-batch-python/src/carddemo_batch/models/transaction.py
"""Transaction model - migrated from COBOL copybook CVTRA05Y.cpy (350 bytes)."""

from decimal import Decimal

from sqlalchemy import Numeric, String
from sqlalchemy.orm import Mapped, mapped_column

from carddemo_batch.models.base import Base


class Transaction(Base):
    """
    Transaction entity representing the TRANSACT VSAM file.
    
    Migrated from COBOL copybook CVTRA05Y.cpy.
    Record length: 350 bytes
    Primary key: TRAN-ID (16 characters)
    """
    __tablename__ = "transact"

    tran_id: Mapped[str] = mapped_column(String(16), primary_key=True)
    tran_type_cd: Mapped[str] = mapped_column(String(2), nullable=True)
    tran_cat_cd: Mapped[int] = mapped_column(nullable=True)
    tran_source: Mapped[str] = mapped_column(String(10), nullable=True)
    tran_desc: Mapped[str] = mapped_column(String(100), nullable=True)
    tran_amt: Mapped[Decimal] = mapped_column(Numeric(12, 2), nullable=True)
    tran_merchant_id: Mapped[str] = mapped_column(String(9), nullable=True)
    tran_merchant_name: Mapped[str] = mapped_column(String(50), nullable=True)
    tran_merchant_city: Mapped[str] = mapped_column(String(50), nullable=True)
    tran_merchant_zip: Mapped[str] = mapped_column(String(10), nullable=True)
    tran_card_num: Mapped[str] = mapped_column(String(16), nullable=True)
    tran_orig_ts: Mapped[str] = mapped_column(String(26), nullable=True)
    tran_proc_ts: Mapped[str] = mapped_column(String(26), nullable=True)

    def __repr__(self) -> str:
        return f"Transaction(tran_id={self.tran_id}, amt={self.tran_amt})"

File: carddemo-java/src/main/java/com/cardemo/entity/Transaction.java
package com.cardemo.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.Table;
import java.math.BigDecimal;

/**
 * Migrated from VSAM TRANSACT (KSDS with AIX on card number) / Copybook CVTRA05Y (350-byte FB records).
 * COBOL: TRAN-RECORD
 */
@Entity
@Table(name = "transactions", indexes = {
    @Index(name = "idx_tran_card_num", columnList = "tran_card_num")
})
public class Transaction {

    /** TRAN-ID PIC X(16) */
    @Id
    @Column(name = "tran_id", length = 16, nullable = false)
    private String tranId;

    /** TRAN-TYPE-CD PIC X(02) */
    @Column(name = "tran_type_cd", length = 2)
    private String tranTypeCd;

    /** TRAN-CAT-CD PIC 9(04) */
    @Column(name = "tran_cat_cd")
    private Integer tranCatCd;

    /** TRAN-SOURCE PIC X(10) */
    @Column(name = "tran_source", length = 10)
    private String tranSource;

    /** TRAN-DESC PIC X(100) */
    @Column(name = "tran_desc", length = 100)
    private String tranDesc;

    /** TRAN-AMT PIC S9(09)V99 */
    @Column(name = "tran_amt", precision = 11, scale = 2)
    private BigDecimal tranAmt;

    /** TRAN-MERCHANT-ID PIC 9(09) */
    @Column(name = "tran_merchant_id")
    private Long tranMerchantId;

    /** TRAN-MERCHANT-NAME PIC X(50) */
    @Column(name = "tran_merchant_name", length = 50)
    private String tranMerchantName;

    /** TRAN-MERCHANT-CITY PIC X(50) */
    @Column(name = "tran_merchant_city", length = 50)
    private String tranMerchantCity;

    /** TRAN-MERCHANT-ZIP PIC X(10) */
    @Column(name = "tran_merchant_zip", length = 10)
    private String tranMerchantZip;

    /** TRAN-CARD-NUM PIC X(16) */
    @Column(name = "tran_card_num", length = 16)
    private String tranCardNum;

    /** TRAN-ORIG-TS PIC X(26) */
    @Column(name = "tran_orig_ts", length = 26)
    private String tranOrigTs;

    /** TRAN-PROC-TS PIC X(26) */
    @Column(name = "tran_proc_ts", length = 26)
    private String tranProcTs;

    public Transaction() {
    }

    public String getTranId() {
        return tranId;
    }

    public void setTranId(String tranId) {
        this.tranId = tranId;
    }

    public String getTranTypeCd() {
        return tranTypeCd;
    }

    public void setTranTypeCd(String tranTypeCd) {
        this.tranTypeCd = tranTypeCd;
    }

    public Integer getTranCatCd() {
        return tranCatCd;
    }

    public void setTranCatCd(Integer tranCatCd) {
        this.tranCatCd = tranCatCd;
    }

    public String getTranSource() {
        return tranSource;
    }

    public void setTranSource(String tranSource) {
        this.tranSource = tranSource;
    }

    public String getTranDesc() {
        return tranDesc;
    }

    public void setTranDesc(String tranDesc) {
        this.tranDesc = tranDesc;
    }

    public BigDecimal getTranAmt() {
        return tranAmt;
    }

    public void setTranAmt(BigDecimal tranAmt) {
        this.tranAmt = tranAmt;
    }

    public Long getTranMerchantId() {
        return tranMerchantId;
    }

    public void setTranMerchantId(Long tranMerchantId) {
        this.tranMerchantId = tranMerchantId;
    }

    public String getTranMerchantName() {
        return tranMerchantName;
    }

    public void setTranMerchantName(String tranMerchantName) {
        this.tranMerchantName = tranMerchantName;
    }

    public String getTranMerchantCity() {
        return tranMerchantCity;
    }

    public void setTranMerchantCity(String tranMerchantCity) {
        this.tranMerchantCity = tranMerchantCity;
    }

    public String getTranMerchantZip() {
        return tranMerchantZip;
    }

    public void setTranMerchantZip(String tranMerchantZip) {
        this.tranMerchantZip = tranMerchantZip;
    }

    public String getTranCardNum() {
        return tranCardNum;
    }

    public void setTranCardNum(String tranCardNum) {
        this.tranCardNum = tranCardNum;
    }

    public String getTranOrigTs() {
        return tranOrigTs;
    }

    public void setTranOrigTs(String tranOrigTs) {
        this.tranOrigTs = tranOrigTs;
    }

    public String getTranProcTs() {
        return tranProcTs;
    }

    public void setTranProcTs(String tranProcTs) {
        this.tranProcTs = tranProcTs;
    }
}
